### PR TITLE
[INLONG-5055][Dataproxy] Fix NPE error of Pulsar client

### LINKED ISSUE #5055: [Bug][Dataproxy] Data won't send due to pulsar client error
### What happened

Data won't send because of pulsar cilent error:
image

Debug and analysis shows that it originates from a NP problem:

image

### What you expected to happen

Send data successfully and no error

### How to reproduce

Fullink configuration that goes through dp, e.g. file -> iceberg

### Environment

_No response_

### InLong version

master

### InLong Component

InLong DataProxy

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by woofyzhao**
- Fixes #5055

### Motivation

Data won't send because of pulsar client error:
image

Debug and analysis shows that it originates from a NP problem:

image

### Modification
Eager initialization and use clear instead of nulling.

### CODE REVIEW

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/pulsar/PulsarClientService.java [hunk 1]
```diff
@@ -534,7 +527,7 @@ class TopicProducerInfo {
         private volatile Boolean isFinishInit = false;
 
         public TopicProducerInfo(PulsarClient pulsarClient, int sinkThreadPoolSize,
-                                 String topic) {
+                String topic) {
```
Comment (healchow) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/pulsar/PulsarClientService.java:530]: Those lines can be merged into one line.

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/pulsar/PulsarClientService.java [hunk 2]
```diff
@@ -402,19 +395,17 @@ public Map<String, List<TopicProducerInfo>> getProducerInfoMap() {
...
-                    logger.error("destroy pulsarClient error in PulsarSink, PulsarClientException {}",
-                            e.getMessage());
-                } catch (Exception e) {
-                    logger.error("destroy pulsarClient error in PulsarSink, ex {}", e.getMessage());
-                }
+        for (PulsarClient pulsarClient : pulsarClients.values()) {
+            try {
+                pulsarClient.shutdown();
+            } catch (PulsarClientException e) {
+                logger.error("destroy pulsarClient error in PulsarSink, PulsarClientException {}",
```
Comment (healchow) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/pulsar/PulsarClientService.java:402]: The error log should print all exception stacks.
For example:
```java
logger.error("destroy PulsarClient error in PulsarSink: ", e);
```

**Review by pocozh — APPROVED** [review]
LGTM

**Review by healchow — APPROVED** [review]

**Review by luchunliang — APPROVED** [review]
LGTM

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**